Assets/Mirror/Runtime/NetworkBehaviour.cs

```diff
@@ -685,24 +685,22 @@ internal bool IsDirty()
         /// <returns>True if data was written.</returns>
         public virtual bool OnSerialize(NetworkWriter writer, bool initialState)
         {
+            bool objectWritten = false;
             if (initialState)
```

Comment:
```suggestion
            // if initialState: write all SyncVars.
            // otherwise write dirtyBits+dirty SyncVars//
            if (initialState)
```